### ISSUE TITLE
Roulette On 2018 Client

### ISSUE THREAD

**Issue by admkakaroto** [rathena/rathena#3266]
* **rAthena Hash**: https://github.com/rathena/rathena/commit/e7fd6aa5bdfac454996c0c1bc720c7cf02022a36
* **Client Date**: 2018-05-30
* **Server Mode**: RE
* **Description of Issue**: Roulette on client 2018 is crashing the game after clicking on it.
it is possible that there is some problem of interpreting packages from the emulator to the client. the packet structure may be malfunctioning or incorrect for new clients.
  * How to Reproduce: just click the roulette icon.
* **Modifications that may affect results**: you must apply the diff: restore roulette on the nemo patcher.
* Refer: https://github.com/HerculesWS/Hercules/commit/8ce6947f6754c797e0557e54af42223212028bb0
Thanks: @sctnightcore

**Comment by Everade** [comment]
Roulette itself works fine, it's more likely your restore diff which crashes your game ;)

**Comment by admkakaroto** [comment]
@Everade you use 2018 client?

**Comment by Everade** [comment]
yes

**Comment by admkakaroto** [comment]
Which patcher did you use? 4144 or secretz?

**Comment by sctnightcore** [comment]
Maybe : https://github.com/HerculesWS/Hercules/pull/2095/commits/8ce6947f6754c797e0557e54af42223212028bb0

**Comment by admkakaroto** [comment]
@sctnightcore has to convert to rA.

**Comment by sctnightcore** [comment]
Maybe .... idk c++
**Clif.cpp**
```
/**
 * Generates information to be displayed
 * @param fd
 * @param sd
 */
void clif_parse_RouletteInfo(int fd, struct map_session_data* sd)
{
	unsigned short i, j, count = 0;
	int len = 8 + (42 * 8);

	nullpo_retv(sd);

	if (!battle_config.feature_roulette) {
		clif_messagecolor(&sd->bl,color_table[COLOR_RED],msg_txt(sd,1497),false,SELF); //Roulette is disabled
		return;
	}

	WFIFOHEAD(fd,len);
	WFIFOW(fd,0) = 0xa1c;
	WFIFOW(fd,2) = len;
	WFIFOL(fd,4) = 1; // serial

	for(i = 0; i < MAX_ROULETTE_LEVEL; i++) {
		for(j = 0; j < MAX_ROULETTE_COLUMNS - i; j++) {
			WFIFOW(fd,8 * count + 8) = i;
			WFIFOW(fd,8 * count + 10) = j;
			WFIFOW(fd,8 * count + 12) = rd.nameid[i][j];
			WFIFOW(fd,8 * count + 14) = rd.qty[i][j];
#if PACKETVER >= 20180523
			WFIFOW(fd,8 * count + 18) = 0;
#endif
			count++;
		}
	}

	WFIFOSET(fd,len);
	return;
}
```

**Comment by admkakaroto** [comment]
@sctnightcore crashed again '-'

**Comment by admkakaroto** [comment]
@nornane crash '--'

**Comment by Lemongrass3110** [comment]
Closed in fe7c0a7
